packages/blocks/shuffle/src/shuffle.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useReducer, useRef } from "react";
+import {
+  BlockComponent,
+  useGraphBlockService,
+} from "@blockprotocol/graph/react";
+// eslint-disable-next-line no-restricted-imports
+import Button from "@mui/material/Button";
+import Box from "@mui/material/Box";
+import { ItemList } from "./components/item-list";
+import { ActionType, initialItems, Items, shuffleReducer } from "./reducer";
+
+type BlockEntityProperties = {
+  items: Items;
+};
+
+export const Shuffle: BlockComponent<BlockEntityProperties> = ({
+  graph: {
+    blockEntity: {
+      entityId,
+      properties: { items },
```

Comment:
I don't think it matters for this block, but we should be aware that if `items` is changed externally (e.g. someone edits the list in a different block), changes will not be reflected in this block.

Reply:
I've made it so the block now reacts to external changes in 1151c80b95b319e555d33f85a69adb9a2c8f57c5 and 1151c80b95b319e555d33f85a69adb9a2c8f57c5